ui/app/adapters/generated-item-list.js

```diff
@@ -6,42 +6,111 @@
 import ApplicationAdapter from './application';
 import { task } from 'ember-concurrency';
 import { service } from '@ember/service';
+import { sanitizePath } from 'core/utils/sanitize-path';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+import { tracked } from '@glimmer/tracking';
 
-export default ApplicationAdapter.extend({
-  store: service(),
-  namespace: 'v1',
-  urlForItem() {},
-  dynamicApiPath: '',
+export default class GeneratedItemListAdapter extends ApplicationAdapter {
+  @service store;
+  namespace = 'v1';
 
-  getDynamicApiPath: task(function* (id) {
-    // TODO: remove yield at some point.
-    const result = yield this.store.peekRecord('auth-method', id);
-    this.dynamicApiPath = result.apiPath;
-    return;
-  }),
+  // these items are set within getNewAdapter in path-help service
```

Comment:
NIT: could we rephrase to clarify these items are not set automatically but only after calling the method? Something like:
```suggestion
  // these items are set by calling getNewAdapter in the path-help service.
```